openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -1970,8 +2036,8 @@ impl UhProcessor<'_, TdxBacked> {
         let regs = self.runner.tdx_enter_guest_gps();
         if regs[TdxGp::R10] == 0 {
             // Architectural VMCALL.
-            let result = match VmxExit(regs[TdxGp::R11] as u32) {
-                VmxExit::MSR_WRITE => {
+            let result = match VmxExitBasic(regs[TdxGp::R11] as u16) {
```

Comment:
Does the ABI specify that we can ignore the top 48 bits of this register?

Reply:
yes - the value in R11 when R10 is zero (meaning, look at intel's GHCI spec) means that 31 and 32 (which match the vmx basic exits) are MSR read/write.